java/vector/src/main/java/org/apache/arrow/vector/complex/ListVector.java

```diff
@@ -105,6 +106,23 @@ public void loadFieldBuffers(ArrowFieldNode fieldNode, List<ArrowBuf> ownBuffers
   }
 
   @Override
+  public void loadFieldBuffers(BuffersIterator buffersIterator, ArrowBuf buf) {
+    buffersIterator.next();
+    ArrowBuf offsetsData = buf.slice((int) buffersIterator.offset(), (int) buffersIterator.length());
+    offsets.load(offsetsData);
+    if (buffersIterator.length() == 0) {
+      offsets.allocateNew(1);
+      offsets.zeroVector();
+    }
+    lastSet = (int) buffersIterator.length() / 4 - 1;
+    // this doesn't match arrow spec, but for backward compatibility, doing it in this order to match getBuffers() method
```

Comment:
This patch is based on the way value vectors were originally designed, before the Arrow project was even started. The java implementation of value vectors still works in much the same way. For example, the FieldNode concept is a new addition, and so the java implementation actually does not currently do anything with the information stored in FieldNode. The Value vectors currently do not keep track of the number of values or the number of nulls. The number of values is determined simply by the size of the buffers. The only exception is the BitVector, which does separately keep track of the value count.

So it's true that tests pass, but this method is designed for a more specific use case that may not work in other use cases.